scripts/o3de/ExportScripts/export_source_built_project.py

```diff
@@ -219,7 +219,7 @@ def parse_args(o3de_context: exp.O3DEScriptExportContext):
                                 help="Any additional file patterns located in the project directory. File patterns will be relative to the project path. Will be included in ServerLauncher.")
         parser.add_argument('-pfp', '--project-file-pattern-to-copy', type=str, dest='project_file_patterns_to_copy', action='append',
                                 help="Any additional file patterns located in the project directory. File patterns will be relative to the project path.")
-        parser.add_argument('-bt', '--build-tools', default=True, action='store_true',
+        parser.add_argument('-bt', '--build-tools', default=False, action='store_true',
```

Comment:
If the built tools are not specified to build, how would the export script know where to get it? I think there is more work that needs to be done here. For example, if this is False, then the user must set the '--tools-build-path' argument. If it is true, it will default to the 'build' subfolder as it does now, or it will use whatever is set by the tools-build-path. I think better documentation on this flag is needed as well.

Reply:
The for locating the tools can follow the same logic that the `AssetSystemComponentHelper_<platform>.cpp` files uses to the to launch the AssetProcesor: https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/Platform/Windows/AzFramework/Asset/AssetSystemComponentHelper_Windows.cpp#L65-L92.

It can check for relative paths as follows
```python
# Normal build folder check when using a source engine
f'bin/{config}'
# Binary folder when using tools from an SDK
f'bin/{platform}/{config}/Default'
# Special case for MacOS to locate the App bundle See https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/Platform/Mac/AzFramework/Asset/AssetSystemComponentHelper_Mac.cpp#L44-L48
f'bin/{platform}/{config}/Default/{appName}.app/Contents/MacOS/{appName}'
```

Reply:
The other thing to consider is that when you run the script, you should know whether or not you are coming from a source build or an sdk-installed build by using the manifest.

Reply:
Actually @tkothdev, Can you add the following additions to allow the o3de CLI tools to detect whether they are running in an SDK layout vs a source layout?
1. Update the [engine.json.in](https://github.com/o3de/o3de/blob/development/cmake/install/engine.json.in#L1-L15) that we [generate](https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Install_common.cmake#L526) when creating the SDK layout to include an `"sdk_engine": true` field within it.
2. We can then add a helper function to manifest.py such as `is_sdk_engine` that queries the "sdk_engine" field
```python
def is_sdk_engine(engine_name: str = None,
                           engine_path: str or pathlib.Path = None) -> bool:
    """
    Queries if the engine is an SDK engine that was created via the CMake install command
    :param engine_name: Name of a registered engine to lookup in the ~/.o3de/o3de_manifest.json file
    :param engine_path: the path to the engine
    :return True if the engine.json contains an "sdk_engine" field which is True.
    """
    engine_json_data = get_engine_json_data(engine_name, engine_path)
    if engine_json_data:
        return engine_json_data.get('sdk_engine', False)

    return False
```